### PR TITLE
apps/questions: fixing poll buttons

### PR DESCRIPTION
**Pull request by khamui**
In Question.jsx a bug around 'showing preliminary results' button is fixed
Additionally following bugs to go back 'to poll' and 'change vote' buttons
are also fixed.

Fixes #2106 (a4-opin repo though)

### CODE REVIEW

**Review by fuzzylogic2000 — CHANGES_REQUESTED** [review]
Very nice! 
I was only wondering if there isn't more code to remove? We must have at least tried to put that into a link? 
And even though it says "button" in all the variables, it was meant to be a link like in a+: https://aplus-dev.liqd.net/liqd-tests/projects/public-poll/
Also, if you want to, you can clean up the code even more! I mean the other parts of it. (And then when it's all nice and clean, we could even port the code to a+ and mB to have it nice and clean everywhere. Or no, I will think about how to use the a4-code again.)

**Comment by fuzzylogic2000** [comment]
Well, actually it may be best to clean and make it nice in a+ and then use that here as well... Let's talk about that!

File: adhocracy4/polls/static/polls/Question.jsx [hunk 1]
```diff
@@ -135,17 +135,21 @@ class Question extends React.Component {
...
-      }
-    } else {
-      toggleShowResultButtonText = django.gettext('Show preliminary results')
-    }
+    const { showResult, selectedChoices: { length: isVoted } } = this.state
+    const buttonText = !showResult
+      ? django.gettext('Show preliminary results')
+      : isVoted
+        ? django.gettext('Change vote')
+        : django.gettext('To poll')
```
Comment (fuzzylogic2000) [inline adhocracy4/polls/static/polls/Question.jsx:143]: Cool! That reads very well!

**Review by fuzzylogic2000 — COMMENTED** [review]
But it's meant to be a link. :grimacing:

**Comment by khamui** [comment]
> Very nice!
> I was only wondering if there isn't more code to remove? We must have at least tried to put that into a link?
> And even though it says "button" in all the variables, it was meant to be a link like in a+: https://aplus-dev.liqd.net/liqd-tests/projects/public-poll/
> Also, if you want to, you can clean up the code even more! I mean the other parts of it. (And then when it's all nice and clean, we could even port the code to a+ and mB to have it nice and clean everywhere. Or no, I will think about how to use the a4-code again.)

I see. Technically it is a button with the appearance of a link right? Because the click event calls a method, rather than following an href. To be meaningful i would avoid to call a method in an <a> element. Rather we restyle a button.

**Comment by khamui** [comment]
Styles adjustments needed for this is in the following PR (a4-opin). Currently only affecting a4-opin, since its used there only.

https://github.com/liqd/a4-opin/pull/2162

**Review by fuzzylogic2000 — APPROVED** [review]
Yeah!

**Pull request merged**